src/main/java/me/ryanhamshire/GriefPrevention/events/PlayerAbandonClaimEvent.java

```diff
@@ -0,0 +1,50 @@
+package me.ryanhamshire.GriefPrevention.events;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import me.ryanhamshire.GriefPrevention.Claim;
+
+public class PlayerAbandonClaimEvent extends Event implements Cancellable {
```

Comment:
I didn't know your plans about revisiting claim permission logic and I'm sorry, I think it's a good idea but now it's not like that (I think...) : I only use the duplicate logic which already were there because I've found that the methods `Claim#allowSomething(Player, [Material])` are more "powerful" that the method `Claim#hasExplicitPermission(Player, ClaimPermission)` because these methods use the other one so that means that setting an event at these methods allows us to do more customizations and therefore I did it. I hope you understand more why I worked that way...

About the `PlayerAbandonClaimEvent` (I'll rename it) I didn't think about /abandonall, I didn't look for each ways to unclaim because I was thinking that it was everytime with the same method so I'll fix it.
But I know that ClaimDeletedEvent is better and handles (I think) everything but as I said, it's fired to late, we can't cancel it so I find it not so useful (and personally I need to cancel some abandons)

Reply:
I assume you meant to respond to the main review, so I'll reply there.